test/ZNSStringResolver.test.ts

```diff
@@ -0,0 +1,464 @@
+  describe("One campaign for everybody", () => {
+    let zeroVault : SignerWithAddress;
+    let operator : SignerWithAddress;
+    let user : SignerWithAddress;
+    let deployAdmin : SignerWithAddress;
+    let deployer : SignerWithAddress;
+    let admin : SignerWithAddress;
+
+    const env = "dev";
+
+    let stringResolver : ZNSStringResolver;
+    let registry : ZNSRegistry;
+    let campaign : DeployCampaign<
+    HardhatRuntimeEnvironment,
+    SignerWithAddress,
+    DefenderRelayProvider,
+    IZNSContracts
+    >;
+    let rootRegistrar : ZNSRootRegistrar;
+
+    let userBalance : bigint;
+
+    const uri = "https://example.com/817c64af";
+    const domainName = "lox";
+    const domainNameHash = hashDomainLabel(domainName);
+
+    let mongoAdapter : MongoDBAdapter;
+
+    before(async () => {
+
+      [
+        deployer,
+        zeroVault,
+        user,
+        deployAdmin,
+        admin,
+        operator,
+      ] = await hre.ethers.getSigners();
+
+      const campaignConfig = await getConfig({
+        deployer: deployer as unknown as SignerWithAddress,
+        governors: [deployAdmin.address],
+        admins: [admin.address],
+        zeroVaultAddress: zeroVault.address,
+        env,
+      });
+
+      campaign = await runZnsCampaign({
+        config: campaignConfig,
+      });
+
+      let meowToken : MeowTokenMock;
+      let treasury : ZNSTreasury;
+
+      ({ stringResolver, registry, meowToken, treasury, rootRegistrar, dbAdapter: mongoAdapter } = campaign);
+
+
+      userBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(user.address, userBalance);
+      await meowToken.connect(user).approve(await treasury.getAddress(), ethers.MaxUint256);
+    });
+
+    after(async () => {
+      await mongoAdapter.dropDB();
+    });
+
+    it("Should not let initialize the contract", async () => {
+      await expect(
+        stringResolver.initialize(
+          await campaign.state.contracts.accessController.getAddress(),
+          await registry.getAddress(),
+        )
+      ).to.be.revertedWith(
+        INITIALIZED_ERR
+      );
+    });
+
+    it("Should correctly attach the string to the domain", async () => {
+
+      await rootRegistrar.connect(user).registerRootDomain(
+        domainName,
+        ethers.ZeroAddress,
+        uri,
+        distrConfigEmpty,
+        paymentConfigEmpty,
+      );
+      await stringResolver.connect(user).setString(domainNameHash, "hippopotamus");
+
+      expect(
+        await stringResolver.resolveDomainString(domainNameHash)
+      ).to.eq(
+        "hippopotamus"
+      );
+    });
+
+    it("Should setRegistry() using ADMIN_ROLE and emit an event", async () => {
+      await expect(
+        stringResolver.connect(admin).setRegistry(admin.address)
+      )
+        .to.emit(stringResolver, "RegistrySet")
+        .withArgs(admin.address);
+
+      expect(await stringResolver.registry()).to.equal(admin.address);
```

Comment:
since you have a single state (a single deployment of ZNS), reset the Registry here to the proper address, otherwise other tests would fail. registry needs to be correct in order for next tests to work properly